Feature/auth_jwt

vibhu0100 opened the pull request.

vibhu0100 committed:
Add files via upload

vibhu0100 committed:
Add files via upload

nehanakrani-simformsolutions commented on app/controllers/application_controller.rb:
```diff
@@ -0,0 +1,46 @@
+    def logged_in?
+    !!logged_in_user
+    end
+
```
Check indentation

nehanakrani-simformsolutions commented on app/controllers/application_controller.rb:
```diff
@@ -0,0 +1,46 @@
+
+                
+            end
+        end
```
Check indentation

nehanakrani-simformsolutions commented on app/controllers/application_controller.rb:
```diff
@@ -0,0 +1,46 @@
+        if auth_header
+            token=auth_header.split(' ')[1]
+            begin
+                JWT.decode(token, 'secret',true,algorith: 'HS256')
```
`token, secret, true, algorith: 'HS256'` make sure after coma space will there in everywhere

nehanakrani-simformsolutions commented on app/controllers/application_controller.rb:
```diff
@@ -0,0 +1,46 @@
+  end
+
+  def auth_header
+    # { Authorization: 'Bearer <token>' }
```
make sure to remove unused code

nehanakrani-simformsolutions commented on app/controllers/application_controller.rb:
```diff
@@ -0,0 +1,46 @@
+    # { Authorization: 'Bearer <token>' }
+    request.headers['Authorization']
+  end
+
```
Why you add extra spaces remove it

nehanakrani-simformsolutions commented on app/controllers/application_controller.rb:
```diff
@@ -0,0 +1,46 @@
+    # { Authorization: 'Bearer <token>' }
+    request.headers['Authorization']
+  end
+
+
+
```
Why you add extra spaces

nehanakrani-simformsolutions commented on app/controllers/application_controller.rb:
```diff
@@ -0,0 +1,46 @@
+    before_action :authorized
+
+    def authorized
+        render json: {message: "Please log in"},status: :unauthorized unless logged_in?
```
comma after space missing

nehanakrani-simformsolutions commented on app/controllers/application_controller.rb:
```diff
@@ -0,0 +1,46 @@
+
+    def encode_token(payload)
+    JWT.encode(payload, 'secret')
+  end
```
Check indentation

nehanakrani-simformsolutions commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,47 @@
+    def create
+        @user=User.new(user_params)
+        if @user.save
+            token=encode_token({user_id: @user.id})
```
missing space

nehanakrani-simformsolutions replied:
missing space

nehanakrani-simformsolutions commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,47 @@
+        else
+            render json: {error: "Invalid"}
+        end
+=begin
```
Remove it if not using

nehanakrani-simformsolutions commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,47 @@
+        else
+            render json: {error: "Invalid"}
+        end
+=begin
+        if user
+            if user.password==params[:password]
+                render json: "Authenticted"
+            else
+                render json: "Enter correct password"
+            end
+        else
+            render json: "You are not registered"
+        end
+=end
```
Remove it if not using make sure code will be clean

nehanakrani-simformsolutions commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,47 @@
+        @user=User.find_by(email: params[:email])
+
+        if @user.password.eql?(params[:password])
+            token=encode_token({user_id: @user.id})
```
Missing space

nehanakrani-simformsolutions commented on bin/bundle:
```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+#
+# This file was generated by Bundler.
```
What is use of this file, why you pushed it

nehanakrani-simformsolutions commented on bin/bundle.cmd:
```diff
@@ -0,0 +1,117 @@
+@ruby -x "%~f0" %*
+@exit /b %ERRORLEVEL%
+
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+#
+# This file was generated by Bundler.
+#
+# The application 'bundle' is installed as part of a gem, and
+# this file is here to facilitate running it.
+#
+
+require "rubygems"
+
+m = Module.new do
+  module_function
```
same here, What is used if this file, why you pushed it

nehanakrani-simformsolutions replied:
same as all bin/ folder files any reason to push or you're missing adding git ignore file
NOTE: understand use of `gitignore`

nehanakrani-simformsolutions commented on log/development.log:
```diff
@@ -0,0 +1,184 @@
+  [1m[35m (83.3ms)[0m  [1m[34mSELECT sqlite_version(*)[0m
+  [1m[35m (221.0ms)[0m  [1m[35mCREATE TABLE "schema_migrations" ("version" varchar NOT NULL PRIMARY KEY)[0m
+  [1m[35m (239.2ms)[0m  [1m[35mCREATE TABLE "ar_internal_metadata" ("key" varchar NOT NULL PRIMARY KEY, "value" varchar, "created_at" datetime(6) NOT NULL, "updated_at" datetime(6) NOT NULL)[0m
+  [1m[35m (0.9ms)[0m  [1m[34mSELECT "schema_migrations"."version" FROM "schema_migrations" ORDER BY "schema_migrations"."version" ASC[0m
+Migrating to CreateUsers (20210303064413)
+  [1m[35m (0.8ms)[0m  [1m[34mSELECT sqlite_version(*)[0m
+  [1m[35m (0.3ms)[0m  [1m[34mSELECT "schema_migrations"."version" FROM "schema_migrations" ORDER BY "schema_migrations"."version" ASC[0m
+Migrating to CreateUsers (20210303064413)
+  [1m[36mTRANSACTION (0.2ms)[0m  [1m[36mbegin transaction[0m
+  [1m[35m (2.1ms)[0m  [1m[35mCREATE TABLE "users" ("id" integer PRIMARY KEY AUTOINCREMENT NOT NULL, "name" varchar, "email" varchar, "password" varchar, "created_at" datetime(6) NOT NULL, "updated_at" datetime(6) NOT NULL)[0m
+  [1m[36mActiveRecord::SchemaMigration Create (1.1ms)[0m  [1m[32mINSERT INTO "schema_migrations" ("version") VALUES (?)[0m  [["version", "20210303064413"]]
+  [1m[36mTRANSACTION (196.9ms)[0m  [1m[36mcommit transaction[0m
+  [1m[36mActiveRecord::InternalMetadata Load (0.7ms)[0m  [1m[34mSELECT "ar_internal_metadata".* FROM "ar_internal_metadata" WHERE "ar_internal_metadata"."key" = ? LIMIT ?[0m  [["key", "environment"], ["LIMIT", 1]]
```
Never push your log file in GitHub

Use cmd line to push code not drag and drop to add files in such case unwanted files may push
Make sure you add gitignore and remove all unwanted files.

vibhu0100 replied:
okay

nehanakrani-simformsolutions commented:
- Your commit name, title and desc should  be informatic
- code indentation must be proper and used in two space conversation default for .rb file
- Definition name must be clear
- Must be add git ignore file in project

nehanakrani-simformsolutions reviewed (commented):
> Must to follow:
> 
> * Your commit name, title and desc should  be informatic
> * code indentation must be proper and used in two space conversation default for .rb file
> * Definition name must be clear
> * Must be add git ignore file in project

vibhu0100 committed:
Delete bin directory

vibhu0100 committed:
Delete development.sqlite3

Delete log directory

Formatted Code

Update application_controller.rb

Update users_controller.rb

vibhu0100 committed:
Delete development.sqlite3

vibhu0100 committed:
Delete log directory

vibhu0100 committed:
Formatted code

vibhu0100 committed:
Formatted code

vibhu0100 committed:
Formatted Code

vibhu0100 committed:
Update application_controller.rb

vibhu0100 committed:
Update users_controller.rb

vibhu0100 committed:
Merge branch 'feature/auth' of https://github.com/vibhu0100/User_authentication_api into feature/auth
